src/compiler/checker.ts

```diff
@@ -16392,10 +16392,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // [...X[]] is equivalent to just X[]
             return readonly ? globalReadonlyArrayType : globalArrayType;
         }
-        const memberIds = mapDefined(namedMemberDeclarations, node => node ? getNodeId(node) : undefined);
         const key = map(elementFlags, f => f & ElementFlags.Required ? "#" : f & ElementFlags.Optional ? "?" : f & ElementFlags.Rest ? "." : "*").join() +
             (readonly ? "R" : "") +
-            (memberIds.length ? "," + memberIds.join(",") : "");
+            (some(namedMemberDeclarations, node => !!node) ? "," + map(namedMemberDeclarations, node => node ? getNodeId(node) : "_").join(",") : "");

```

Comment:
Is there a reason to precheck rather than just doing the map and letting it give the empty string via returning nothing?

Reply:
Hmm, not rly - this only keeps the cache key shorter if there are no named members.

Reply:
Oh, I misread this and didn't notice that this was adding a conditional prefix. Just seemed less nice to double iterate but it's fine.

Reply:
I debugged this down in a moment between things so I didn't exactly give it an extended thought at that moment. It was previously added conditionally and I didn't spend a second thinking if this would still be needed after things change. 

Since you called it out, I re-evaluated it and I think it is, indeed, redundant now. So I would be more than OK with dropping this. Unlabeled tuples are probably more common than labeled ones though but I doubt that the cache key size would impact perf in a significant way.

Reply:
All good, I'd just leave it as it's not a regression.